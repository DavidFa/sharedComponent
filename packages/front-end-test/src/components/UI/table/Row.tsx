import React from 'react'
import styled from 'styled-components';

const Tr = styled.tr`
color: inherit;
display: table-row;
vertical-align: middle;
outline: 0px;
`

// type RowProps = {
//     postId?: number;
//     rowClickHandler?: (postId: number) => void;
// }

const Row: React.FC = ({ children }) => {

    // let row;
    // if (postId && rowClickHandler) {
    //     row = <Tr onClick={() => rowClickHandler(postId)}>{children}</Tr>
    // } else {
    //     row = <Tr>{children}</Tr>;
    // }
    return <Tr>{children}</Tr>;
}

export default Row;
