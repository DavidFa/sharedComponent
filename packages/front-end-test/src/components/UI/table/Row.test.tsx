import Row from "./Row";
import { screen, render } from "@testing-library/react";
import renderer from 'react-test-renderer';

describe("Row", () => {
    describe("Snapshot Test", () => {
        it('should render with td', () => {
            const component = renderer.create(
                <Row>
                <td>Table Cell Data</td>
              </Row>
            )
            expect(component.toJSON()).toMatchSnapshot();
        })
    });


  describe("Dom Test", () => {
    const table = document.createElement("table");
    const tbody = document.createElement("tbody");
    const tableContainer = document.body.appendChild(table).appendChild(tbody);

    it("should render with td", () => {
      const { container } = render(
        <Row>
          <td>Table Cell Data</td>
        </Row>,
        {
          container: tableContainer,
        }
      );

      const row = screen.queryByRole("row");
      expect(row).toBeInTheDocument();

      expect(container.firstChild).toMatchInlineSnapshot(`
        <tr
          class="sc-bdfBQB bGyyoG"
        >
          <td>
            Table Cell Data
          </td>
        </tr>
      `);
    });
  });
});
