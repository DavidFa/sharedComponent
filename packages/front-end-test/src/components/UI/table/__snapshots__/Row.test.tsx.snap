// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Snapshot Test should render with td 1`] = `
<tr
  className="sc-bdfBQB bGyyoG"
>
  <td>
    Table Cell Data
  </td>
</tr>
`;
